Parse/ParseClientConfiguration.h

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <Parse/PFConstants.h>
+
+/*!
+ This represents the entirety of a parse configuration
+ */
+@interface ParseClientConfiguration : NSObject<NSCopying>
+
+@property (nonatomic, copy) NSString *applicationId;
+@property (nonatomic, copy) NSString *clientKey;
+
+@property (nonatomic, assign, getter=isLocalDatastoreEnabled)BOOL localDatastoreEnabled;
+
+@property (nonatomic, copy) NSString *applicationGroupIdentifier;
+@property (nonatomic, copy) NSString *containingApplicationBundleIdentifier;
+
```

Comment:
Mark `init` and `new` as unavailable, forcing initialization purely via designated constructor.